### ISSUE TITLE
App architecture for large/enterprise apps

### ISSUE THREAD

**Issue by mumairofficial** [toddmotto/angular-architecture#1]
Hi Todd,

I am learning application architecture for complex applications such as Dashboard (Management Systems);
my question is what folder and file structures would you suggest? when we have following scenario:

**Containers**: 
- Users management
- Inventory management
- Dashboard configurations
- so on

and **Components** may be some used common (headers, footer, bread crumb, left navigation) etc. or some related to containers only.

**Comment by StevenFewster** [comment]
The most useful way to contribute might be to fork, and show proposed solutions - the difficulty being how to keep it generic enough to be useful to all projects, and specific enough to provide a guide to all.

So for user management, we're going to need a global and local proposition for Guards, Dashboard configs sound like layout, and layout might also have to be repeated again at component level.  Could we, as a group, get this going?

**Comment by mumairofficial** [comment]
Yes, only challenge comes when there are multiple modules lets say lazy loaded modules and we have some root level common components and some module/container specific components.
